### ISSUE TITLE
Can't share screen after sharing it once (DOMException: Invalid state)

### ISSUE THREAD

**Issue by nsharma1396** [muaz-khan/RTCMultiConnection#703]
Hi,
Excellent work on the library! Works like a charm!
But, recently I got stuck with a problem while using your library with react where whenever I am trying to share my screen the second time on Chrome, I get an error:

`DOMException: Invalid state {audio: false, video: {…}, isScreen: true}`

I understand that this error is due to the **chromeMediaSourceId** being reused from the last screen share (while chrome changes it on every stream) and so I tried to manually capture the screen using Screen-Capturing.js (by setting the **sourceId** to **null**) and I did get the screen and audio media stream source objects from the function even the second time. But I was unable to properly attach the streams to the connection.

I have tried several ways to fix the problem but can't seem to get them working.

Is there any way using which I can reset the **chromeMediaSourceId** everytime the screen stream is requested while still maintaining the normal flow of your library ?
If not, then is there any alternative that I can follow.

Thank you.

**Comment by muaz-khan** [comment]
You can try [`connection.resetScreen`](https://www.rtcmulticonnection.org/docs/resetScreen/).

Personally I prefer `replaceTrack`, as I used in this demo:
* https://rtcmulticonnection.herokuapp.com/demos/video-and-screen-sharing.html

`connection.resetScreen` will allow me capture screen many times; and `replaceTrack` trick will make sure that screen is shared/removed seamlessly i.e. without any issues or failures.

```javascript
sourceId = null;
getScreenConstraints(function(error, screen_constraints) {
    navigator.mediaDevices.getUserMedia({
        video: screen_constraints
    }).then(function(stream) {
        replaceTrack(stream);
       // addStreamStopListener(stream, callback);
       // preview.srcObject = stream;
       // etc.
    })
});
```

I'm using same trick on `dashboard` demo:
* https://rtcmulticonnection.herokuapp.com/demos/dashboard/

Dashboard demo tries to open a hidden `video-track` (via `canvas.captureStream`) and use it to replaceTrack whenever needed.

i.e. if you share screen, it will replace only canvas-captured-screen' track; and if you stop sharing then it will disable that track. i.e. it doesn't affects camera or other tracks.

I'm using WebRTC data channels along with replaceTrack because I need to tell remote parties that show screen if it is shared; otherwise hide it.

If you're unable to use similar technique then please let me know. Then I'll publish a detailed comment along with code snippets. i.e. how to use replaceTrack etc.

**Comment by nsharma1396** [comment]
Hi @muaz-khan,

Thank you so much for the response. Apologies for the delay in replying.

I used **connection.resetscreen** and that did remove the DOMException but on trying to stop sharing the screen and getting the video stream back, I started getting errors.
So I shifted to **replaceTrack** and **resetTrack** and they did work perfectly. I was able to seamlessly shift between the screen track and the video track but had to use the replaceTrack api a bit differently like so:

```
// stream is the screen stream we received from **getUserMedia**
const videoTrack = stream.getVideoTracks()[0];
videoTrack.onended = () => {
    this.connection.resetTrack();
}

this.connection.getAllParticipants().forEach( participant =>
     this.connection.replaceTrack(videoTrack, participant, true)
);
```
But, as you said that this approach should not affect the camera track

>  if you share screen, it will replace only canvas-captured-screen' track; and if you stop sharing then it will disable that track. i.e. it doesn't affects camera or other tracks.

So, I was wondering whether I was using the api in the best possible way my code is basically **replacing** the **camera track with the screen track** every time and then switching back to the original track or is there some better way for the same.

Thank you.
